src/MOI_wrapper.jl

```diff
@@ -2512,13 +2512,45 @@ function _dual_multiplier(model::Optimizer)
     return MOI.get(model, MOI.ObjectiveSense()) == MOI.MIN_SENSE ? 1.0 : -1.0
 end
 
+"""
+    _farkas_variable_dual(model::Optimizer, col::Cint)
+
+Return a Farkas dual associated with the variable bounds of `col`.
+
+Gurobi computes the Farkas dual as:
+
+    ā * x = λ' * A * x <= λ' * b = -β + sum(āᵢ * Uᵢ | āᵢ < 0) + sum(āᵢ * Lᵢ | āᵢ > 0)
+
+The Farkas dual of the variable is ā, and it applies to the upper bound if ā < 0,
+and it applies to the lower bound if ā > 0.
+"""
+function _farkas_variable_dual(model::Optimizer, col::Cint)
+    valueP = Ref{Cint}()
+    ret = GRBgetintattr(model, "NumConstrs", valueP)
+    _check_ret(model, ret)
+    ā = 0.0
+    λ, a = Ref{Cdouble}(), Ref{Cdouble}()
+    for row = 1:valueP[]
+        ret = GRBgetdblattrelement(model, "FarkasDual", Cint(row - 1), λ)
+        _check_ret(model, ret)
+        ret = GRBgetcoeff(model, Cint(row - 1), col, a)
```

Comment:
Looping through all entries in the constraint matrix is *way* too expensive to do by default.

Reply:
Ah, I missed this: https://www.gurobi.com/documentation/9.0/refman/c_getvars.html

Reply:
using `GRBgetvars` should be faster.